src/index.js

```diff
@@ -185,7 +185,11 @@ export default async function fetch(url, options_) {
 				});
 			}
 
-			let body = pump(response_, new PassThrough(), reject);
+			let body = pump(response_, new PassThrough(), err => {
```

Comment:
Sorry for the wait, this PR looks fine to me, but I think a few things can be improved:
 
- If the change on `pump()` error handling is necessary for your tests to pass, then we should probably change all of the instances.